### PR TITLE
Store both IPv4 and IPv6 addresses in the crypto connection

### PR DESCRIPTION
**Pull request by kurnevsky**

### CODE REVIEW

File: src/toxcore/net_crypto/crypto_connection.rs [hunk 1]
```diff
@@ -234,6 +236,35 @@ impl RecvPacket {
     }
 }
 
+/// UDP address of a connection with the time when last UDP packet was received
+#[derive(Clone, Debug, PartialEq)]
+pub struct ConnectionAddr<T: Into<SocketAddr> + Copy> {
+    /// Address to send UDP packets directly to the peer
+    pub addr: T,
+    /// Time when last UDP packet was received
+    pub received_time: Instant,
```
Comment (kpp) [inline src/toxcore/net_crypto/crypto_connection.rs]: last_received_time?

File: src/toxcore/net_crypto/crypto_connection.rs [hunk 2]
```diff
@@ -433,9 +464,60 @@ impl CryptoConnection {
         }
     }
 
-    /// Set time when last UDP packet was received to now
-    pub fn update_udp_received_time(&mut self) {
-        self.udp_received_time = Some(clock_now())
+    /// Set UPD address for this connection
+    pub fn set_udp_addr(&mut self, addr: SocketAddr) {
+        match addr {
+            SocketAddr::V4(addr) => self.udp_addr_v4 = Some(ConnectionAddr::new(addr)),
```
Comment (kpp) [inline src/toxcore/net_crypto/crypto_connection.rs]: https://doc.rust-lang.org/std/option/enum.Option.html#method.replace

Comment (kurnevsky) [reply src/toxcore/net_crypto/crypto_connection.rs]: Replace is used to get old value which I don't need. I just need to put a new value there.

Comment (kpp) [reply src/toxcore/net_crypto/crypto_connection.rs]: Too bad Option does not have a method like `::set`.

Comment (kurnevsky) [reply src/toxcore/net_crypto/crypto_connection.rs]: I think it's because `=` does exact same thing :)

Comment (kpp) [reply src/toxcore/net_crypto/crypto_connection.rs]: :D

**Pull request merged**